File: solutions/elixir/acronym/1/lib/acronym.ex
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
      string
      |> String.upcase
      |> String.replace("-", " ")
      |> String.replace("_", "")
      |> String.split(" ")
      |> Enum.map(fn x -> String.at(x, 0) end)
      |> Enum.join
  end
end

File: solutions/javascript/acronym/1/acronym.js
/**
 * Parses a phrase to generate its acronym.
 * 
 * @param {string} phrase - The phrase to parse.
 * @returns {string} The generated acronym.
 */
function parse(phrase) {
  return phrase
    .replace("'", '')
    .match(/^[A-Z]|(?<=[^A-Z])[A-Z]|\b[a-z]/g)
    .join('')
    .toUpperCase();
}

export { parse };
